spec.bs

```diff
@@ -252,19 +252,218 @@ dictionary AuctionAdConfig {
+    "`https`".
+  1. Set |auctionConfig|'s [=auction config/seller=] to |seller|.
+1. Let |decisionLogicUrl| be the result of running the [=URL parser=] on
+  |config|["{{AuctionAdConfig/decisionLogicUrl}}"].
+  1. [=exception/Throw=] a {{TypeError}} if |decisionLogicUrl| is an error, or it is not
+    [=same origin=] with |auctionConfig|'s [=auction config/seller=].
+  1. Set |auctionConfig|'s [=auction config/decision logic url=] to |decisionLogicUrl|.
+1. If |config|["{{AuctionAdConfig/trustedScoringSignalsUrl}}"] [=map/exists=]:
+  1. Let |trustedScoringSignalsUrl| be the result of running the [=URL parser=] on
+    |config|["{{AuctionAdConfig/trustedScoringSignalsUrl}}"].
+  1. [=exception/Throw=] a {{TypeError}} if |trustedScoringSignalsUrl| is an error,
+    or it is not [=same origin=] with |auctionConfig|'s [=auction config/seller=].
+  1. Set |auctionConfig|'s [=auction config/trusted scoring signals url=] to
+    |trustedScoringSignalsUrl|.
+1. If |config|["{{AuctionAdConfig/interestGroupBuyers}}"] [=map/exists=], let |buyers| be a new
+  [=list/is empty|empty=] [=list=].
+  1. [=list/For each=] |buyerString| in |config|["{{AuctionAdConfig/interestGroupBuyers}}"]:
+    1. Let |buyer| be the result of [=parsing an origin=] with |buyerString|. If |buyer| is an
+      error, then [=exception/throw=] a {{TypeError}}. Otherwise, [=list/append=] |buyer| to
+      |buyers|.
+    1. If |isTopLevel| is true and |config|["{{AuctionAdConfig/componentAuctions}}"] [=map/exists=],
+      TODO: |buyer| cannot be a buyer who directly participates in the top level auction, i.e.,
+      |buyer| must be a buyer who won a component auction of
+      |config|["{{AuctionAdConfig/componentAuctions}}"].
+  1. Set |auctionConfig|'s [=auction config/interest group buyers=] to |buyers|.
+1. If |config|["{{AuctionAdConfig/auctionSignals}}"] [=map/exists=], let |auctionConfig|'s
+  [=auction config/auction signals=] be the result of
+  [=serializing a JavaScript value to a JSON string=], given
+  |config|["{{AuctionAdConfig/auctionSignals}}"].
+1. If |config|["{{AuctionAdConfig/sellerSignals}}"] [=map/exists=], let |auctionConfig|'s
+  [=auction config/seller signals=] of be the result of
+  [=serializing a JavaScript value to a JSON string=], given
+  |config|["{{AuctionAdConfig/sellerSignals}}"].
+1. If |config|["{{AuctionAdConfig/directFromSellerSignals}}"] [=map/exists=], let
+  |directFromSellerSignalsPrefix| be the result of running the [=URL parser=] on
+  |config|["{{AuctionAdConfig/directFromSellerSignals}}"].
+  1. [=exception/Throw=] a {{TypeError}} if any of:
+    * |directFromSellerSignalsPrefix| is an error.
+    * |directFromSellerSignalsPrefix| is not [=same origin=] with |auctionConfig|'s
+      [=auction config/seller=].
+    * |directFromSellerSignalsPrefix|'s [=url/query=] is not null.
+  1. TODO: Figure out how to deal with DirectFromSellerSignals.
+1. If |config|["{{AuctionAdConfig/sellerTimeout}}"] [=map/exists=], set |auctionConfig|'s

```

Comment:
We do have a 500ms max (>500ms will be clamped to 500). There's a default 50ms if a timeout is not provided. I didn't mention these because in our implementation, we just copy the value in the given AuctionConfig to the internal config, and do these min/max/default stuff later at different places before being used, instead of at the time of converting AuctionConfig to the internal config.

It's a good question about minimum value. We don't have one for the timeouts currently, but I agree that we should have one set to 1ms since allowing 0ms timeout does not make much sense?